subsys/bootloader/bl_validation/bl_validation.c

```diff
@@ -239,6 +252,42 @@ static bool validate_hash(u32_t fw_src_address, const struct fw_info *fwinfo,
 #endif
 
 
+static bool within(u32_t addr, u32_t start, u32_t end)
```

Comment:
Should we have a unit test for these? We could have a simple one including the .c file to avoid the overhead of moving this out to a separate file.

Reply:
Added a new test for these (bl_validation_unittest). I had to move them into a separate file (bl_validation_internal.h) because bl_validation.c needs a lot of dependencies to build.